lib/active_merchant/billing/gateways/creditcall.rb

```diff
@@ -126,17 +126,22 @@ def add_terminal_details(xml, options={})
 
       def add_card_details(xml, payment_method, options={})
         xml.CardDetails do
-          xml.Manual(type: "ecommerce") do
+          xml.Manual(type: manual_type(options)) do
             xml.PAN payment_method.number
             xml.ExpiryDate exp_date(payment_method)
             xml.CSC payment_method.verification_value unless empty?(payment_method.verification_value)
           end
 
-          if address = options[:billing_address]
-            xml.AdditionalVerification do
-              xml.Address address[:address1]
-              xml.Zip address[:zip]
-            end
+          add_additional_verification(xml, options)
+        end
+      end
+
+      def add_additional_verification(xml, options)
+        return unless (options[:verify_zip] == 'true') || (options[:verify_address] == 'true')
```

Comment:
I guess the more I think about it, with Ruby we could just do:

```ruby
return unless (options[:verify_zip]) || (options[:verify_address])
```

and it would work since the conditional is only checking absence of `nil` or `false`, in which case if it was `true` or `"true"` would both work.